### DISCUSSION TITLE
How to use Next + React 18 with legacy ReactDOM.render()

### DISCUSSION THREAD

**Discussion by chentsulin** [vercel/next.js#36770] [Help]
Considering we have a large-scale monorepo and a lot of Next.js sites, it's almost impossible for us to make all sites React 18 `createRoot` compatible at a time. 

However, using multiple version of React in a monorepo could be a huge mess.

Is there any config can leave some sites using `ReactDOM.render` but still work with React 18?

**Comment by icyJoseph** [comment]
I think you can pass an experimental config, called `reactRoot`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  experimental: {
    reactRoot: false,
  },
};

module.exports = nextConfig;
```

I think that's it, right, could you confirm? 

I am really not sure now, I can't observe anything to make me say 100% that this is the change you need.

**Reply by chentsulin** [reply]
@icyJoseph It doesn't work. The `Root` still be rendered by `hydrateRoot`:

<img width="673" alt="截圖 2022-05-09 下午10 03 46" src="https://user-images.githubusercontent.com/3382565/167429389-e8b56d73-a796-4598-b3d9-3d709f605144.png">

The default value of `reactRoot` seems to be `false` and it will be overridden to `true` when React 18 is detected.
https://github.com/vercel/next.js/blob/e8a8220d49434021f2fb9975a6ec7081043b97b4/packages/next/build/webpack-config.ts#L347-L350

**Reply by icyJoseph** [reply]
I see, I guess upgrading to React 18 and not taking the new rendering, is just about the same as sticking with 17. Next.js seems to be making such a decision. 

Out of curiosity, what were you trying to achieve by upgrading to 18 but keeping the old render method?

**Reply by chentsulin** [reply]
We have more than 20 Next.js sites in our monorepo, and 10% of them are currently blocking by their dependencies. Our best plan is to upgrade these sites to React 18 at the same time but with the old render method.

**Comment by chentsulin** [comment]
found a relevant PR: https://github.com/vercel/next.js/pull/36419
